### PR TITLE
DOC: Fixes to the docs style

### PR DESCRIPTION
**Pull request by datapythonista**
Fixing flake8 errors in documentation pages that are mostly correct. Also changing `setup.cfg` to avoid false positives.

- [X] refs #24173
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -32,10 +32,18 @@ exclude =
...
 [flake8-rst]
 bootstrap =
-    import pandas as pd
     import numpy as np
+    import pandas as pd
+    np  # avoiding error when importing again numpy or pandas
+    pd  # (in some cases we want to do it to show users)
 ignore = E402,  # module level import not at top of file
          W503,  # line break before binary operator
+         # Classes/functions in different blocks can generate those errors
```
Comment (jreback) [inline setup.cfg]: @datapythonista why are you adding these? we *want* to error these, yes?

Comment (datapythonista) [reply setup.cfg]: Not in the documentation. `flake8-rst` merges all the blocks in a single file to validate it. Meaning that if I have a block of code that shows a class definition, if we validate for those errors, we will get an error, as the class definition will be immediately after the last statement of the previous block.

**Comment by jreback** [comment]
thanks @datapythonista

File: setup.cfg (hunk 1 above)
Comment (jreback) [reply setup.cfg]: ok thanks. maybe add what you just wrote in a future PR as a reader would have no idea how this works.

Comment (datapythonista) [reply setup.cfg]: I tried to explain in the comment before skipping the errors, but obviously not clear enough. Will add more context/info in a future PR.

**Pull request merged**